src/Microsoft.AspNetCore.JsonPatch/Internal/DynamicObjectAdapter.cs

```diff
@@ -96,6 +96,11 @@ public class DynamicObjectAdapter : IAdapter
                 return false;
             }
 
+            if (!TryRemove(target, segment, contractResolver, out errorMessage))
```

Comment:
Are there any real cases where `TryRemove` will fail? What are the semantics of the spec when you have a replace operation but there is "nothing to remove"?

Reply:
According to the spec, `replace` should function as a `remove` followed by an `add`. The spec says for the `remove` op:

> The target location MUST exist for the operation to be successful.

So I believe that `replace` will fail if the target location does not already exist. `TryRemove` looks like it verifies the target location exists and will fail if it doesn't.

Reply:
I think I misunderstood what you meant actually. If you mean what should happen if a property exists within the object graph at the location specified by the patch operation, but it's set to null or something like that - I believe the replace operation should still succeed, in that I can't find anything in the spec that explicitly rules against that. I think the present implementation allows for that as well - `TryConvertValue` looks like it succeeds for properties that are reference types and are set to null.

Reply:
ok